### PR TITLE
Respect root dir/common src dir when generating module names

### LINKED ISSUE #5819: rootDir doesn't affect outFile output (and working dir always does)
If my code is under `src/` then the `.d.ts` file produced together with `outFile` declares a module `src/moduleName` and the `.js` file contains for example `System.register("src/moduleName", ...`. Setting `--rootDir src` should IMHO remove the initial `src` component, like it does with `outDir`.

Also, whatever path is given to `tsc -p` also shows up as a prefix in paths output to `outFile`. This means the compiled output depends on the working directory where `tsc` is run, even if it's explicitly given a specific project directory!

### PR DESCRIPTION
**Pull request by weswigham**
Fixes #5819.

The root of the issue was that we weren't combining/separating paths correctly when joining/splitting the common source directory with the filename to generate the module name.

### CODE REVIEW

File: tests/baselines/reference/getEmitOutputSingleFile2.baseline [hunk 1]
```diff
@@ -23,7 +23,7 @@ declare class Foo {
     x: string;
     y: number;
 }
-declare module "inputFile3" {
+declare module "inputfile3" {
```
Comment (mhegazy) [inline tests/baselines/reference/getEmitOutputSingleFile2.baseline]: this should not change... the file name specified is "inputFile3"

File: tests/baselines/reference/commonSourceDir5.js [hunk 2]
```diff
@@ -16,17 +16,17 @@ export var pi = Math.PI;
 export var y = x * i;
 
 //// [concat.js]
-define("B:/baz", ["require", "exports", "A:/bar", "A:/foo"], function (require, exports, bar_1, foo_1) {
```
Comment (mhegazy) [inline tests/baselines/reference/commonSourceDir5.js]: same here, should be "B".

File: tests/baselines/reference/getEmitOutputSingleFile2.baseline (hunk 1 above)
Comment (weswigham) [inline tests/baselines/reference/getEmitOutputSingleFile2.baseline]: This is as a result of the canonicalization now performed on the file name in order to get the whole path (and then trim off the common part).

I suppose I could rework the comparison so not canonicalize and split, but instead compare canonicalized components and join, in order to preserve input casing on case insensitive systems?

**Comment by mhegazy** [comment]
:+1: thanks!

**Pull request merged**